### PR TITLE
task done

### PR DESCRIPTION
**Pull request by taraShulak**
- [DEMO LINK]https://taraShulak.github.io/layout_product-cards/
- [TEST REPORT LINK]https://taraShulak.github.io/layout_product-cards/report/html_report/)

- [ x] there 2 BEM blocks `card` and `stars` each in it own file
- [ x] SCSS Nesting is used for `elements`, `modifiers` and `pseudo-classes`
- [ x] SCSS Variables are used for main values and placed in a **separate file**
- [ x] all `stars--N` modifiers work as expected (Highlight first `N` stars)
- [ x] Code follows all the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -5,9 +5,55 @@
...
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
     <link rel="stylesheet" href="./styles/index.scss">
   </head>
   <body>
-    <h1>Product cards</h1>
+    <main class="main">
+      <div class="card" data-qa="card">
+        <div class="card__photo">
```
Comment (volodymyr-soltys97) [inline src/index.html:16]: This div is redundant
```suggestion
```

File: src/index.html [hunk 2]
```diff
@@ -5,9 +5,55 @@
...
+          <img src="/src/images/imac.jpeg"
+            class="photo"
+            alt="photo">
```
Comment (volodymyr-soltys97) [inline src/index.html:19]:
```suggestion
          <img
            src="/src/images/imac.jpeg"
            class="photo"
            alt="iMac photo"
          >
```

File: src/index.html [hunk 3]
```diff
@@ -5,9 +5,55 @@
...
+        </div>
+        <div class="card__name">
+          APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)
+        </div>
```
Comment (volodymyr-soltys97) [inline src/index.html:23]: Use the semantic tag for text, need to fix it everywhere

File: src/index.html [hunk 4]
```diff
@@ -5,9 +5,55 @@
...
+        <div class="card__price">
+          <div class="card__price-title">
+            Price:
+          </div>
+          <div class="card__price-price">
+            $2,199
+          </div>
+        </div>
+        <div class="card__buy">
```
Comment (volodymyr-soltys97) [inline src/index.html:49]: Redundant
```suggestion
```

File: src/index.html [hunk 5]
```diff
@@ -5,9 +5,55 @@
...
+          <a href="#"
+            data-qa="hover"
+            class="card__link buy">
+            <button class="button">BUY</button>
+          </a>
```
Comment (volodymyr-soltys97) [inline src/index.html:54]:
```suggestion
          <a
            href="#"
            data-qa="hover"
            class="card__link buy"
          >
            BUY
          </a>
```

File: src/styles/index.scss [hunk 6]
```diff
@@ -1,3 +1,133 @@
+$main-accent: #060B35;
+$secondary: #616070;
+$blue-accent: #00ACDC;
+$white: #FFF;
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:4]: Move variables to the separate file

File: src/styles/index.scss [hunk 7]
```diff
@@ -1,3 +1,133 @@
...
+  @include main-font;
+
+  margin: 0;
+  font-style: normal;
+  font-weight: 400;
+}
+
+.main {
+  width: 200px;
+  height: 406px;
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:20]: No need to hardcode the height for the card
```suggestion
```

File: src/styles/index.scss [hunk 8]
```diff
@@ -1,3 +1,133 @@
...
+    height: 134px;
+    padding: 16px 3px;
+  }
+
+  &__name {
+    padding: 24px 0 4px;
+    color: $main-accent;
+    font-size: 12px;
+    font-weight: 500;
+    line-height: 18px; /* 150% */
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:57]: Remove comments everywhere
```suggestion
    line-height: 18px; 
```

File: src/styles/index.scss [hunk 9]
```diff
@@ -1,3 +1,133 @@
...
+    height: 16px;
+    width: 16px;
+    background-image: url(/src/images/star.svg);
+    background-position: center;
+    background-repeat: no-repeat;
+  }
+
+  .stars--4 .stars__star:nth-child(-n + 4) {
+    background-image: url(/src/images/star-active.svg);
+  }
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:132]: Move stars styles to the separate file

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code
This part of the task is incomplete
image
image

File: src/index.html (hunk 1 above)
Comment (Andriy-Shymkiv) [reply src/index.html:16]: no fixed

File: src/index.html [hunk 10]
```diff
@@ -5,9 +5,51 @@
...
   </head>
   <body>
-    <h1>Product cards</h1>
+    <main class="main">
+      <div class="card" data-qa="card">
+        <div class="card__photo">
+          <img
+            src="./images/imac.jpeg"
+            class="photo"
+            alt="photo">
```
Comment (Andriy-Shymkiv) [inline src/index.html:20]:
```suggestion
          <img
            src="./images/imac.jpeg"
            class="photo"
            alt="photo"
          >
```

File: src/index.html (hunk 4 above)
Comment (Andriy-Shymkiv) [reply src/index.html:44]: not fixed

**Review by Andriy-Shymkiv — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 11]
```diff
@@ -5,9 +5,48 @@
...
   <body>
-    <h1>Product cards</h1>
+    <main class="main">
+      <div class="card" data-qa="card">
+        <img
+          src="./images/imac.jpeg"
+          class="card__photo"
+          alt="photo"
+        >
+        <p class="card__name">APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</p>
```
Comment (anastasiiavorobiova) [inline src/index.html:21]:
```suggestion
        <h2 class="card__name">APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</h2>
```

File: src/index.html [hunk 12]
```diff
@@ -5,9 +5,48 @@
...
+        <div class="card__price">
+          <p class="card__price-title">Price:</p>
+          <p class="card__price-price">
+            $2,199
+          </p>
+        </div>
+        <a
+          href="#"
+          data-qa="hover"
+          class="card__link">
```
Comment (anastasiiavorobiova) [inline src/index.html:46]:
```suggestion
          class="card__link"
        >
```

File: src/index.html [hunk 13]
```diff
@@ -5,9 +5,48 @@
...
+          <button class="button">BUY</button>
```
Comment (anastasiiavorobiova) [inline src/index.html:47]:
```suggestion
          BUY
```

File: src/styles/stars.scss [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+  &__star {
+    margin: 0;
+    height: 16px;
+    width: 16px;
+    background-image: url(/src/images/star.svg);
+    background-position: center;
+    background-repeat: no-repeat;
+  }
+
+  .stars--4 .stars__star:nth-child(-n + 4) {
```
Comment (anastasiiavorobiova) [inline src/styles/stars.scss:19]: Consider using the whole CSS from the previous task

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Almost done!

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Make sure to deploy the page

**Comment by taraShulak** [comment]
Знімок екрана 2024-01-21 о 22 06 22
Знімок екрана 2024-01-21 о 22 06 07

**Comment by taraShulak** [comment]
test were good, npm start also was good......why pages does not work?

**Pull request closed without merging**